geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/Part.java

```diff
@@ -175,6 +175,34 @@ public String getString() {
     return CacheServerHelper.fromUTF((byte[]) this.part);
   }
 
+  @MakeNotStatic
+  private static final Map<ByteBuffer, String> CACHED_STRINGS = new ConcurrentHashMap<>();
+
+  private static String getCachedString(byte[] serializedBytes) {
+    ByteBuffer key = ByteBuffer.wrap(serializedBytes);
```

Comment:
Why allocate a ByteBuffer? The main goal here is to eliminate object creation and while this removes some it also adds one.

Reply:
We need some type of wrapper around a byte array so that we can do equals and hashcode on it. In Java equals and hashcode on arrays is not based on the contents of the array but on its identity. I think ByteBuffer is a pretty cheap way to get this. We are using "wrap" so the byte array is not copied. The alternative would be to use a map that supports array keys and calls Arrays.equals and Arrays.hashCode on the keys.

Reply:
I changed it from ByteBuffer to a new ByteArrayKey class that is a little smaller and makes clear why it exists.